src/app/SimpleSubscriptionResumptionStorage.cpp

```diff
@@ -318,44 +374,63 @@ CHIP_ERROR SimpleSubscriptionResumptionStorage::Save(SubscriptionInfo & subscrip
     ScopedNodeId subscriptionNode = ScopedNodeId(subscriptionInfo.mNodeId, subscriptionInfo.mFabricIndex);
     // Load index and update if fabric/node is new
     SubscriptionIndex subscriptionIndex;
-    LoadIndex(subscriptionIndex);
+    LoadIndex(subscriptionIndex, 1);
     bool nodeIsNew = true;
     for (size_t i = 0; i < subscriptionIndex.mSize; i++)
     {
-        if (subscriptionNode == subscriptionIndex.mNodes[i])
+        if (subscriptionNode == subscriptionIndex[i])
         {
             nodeIsNew = false;
             break;
         }
     }
     if (nodeIsNew)
     {
+        if (!subscriptionIndex.mSize)
+        {
+            subscriptionIndex.mNodes = std::unique_ptr<ScopedNodeId[]>(new(std::nothrow) ScopedNodeId[1]);
+            if (!subscriptionIndex.mNodes)
+            {
+                return CHIP_ERROR_NO_MEMORY;
+            }
+        }
+
         if (subscriptionIndex.mSize == CHIP_IM_MAX_NUM_SUBSCRIPTIONS)
         {
             return CHIP_ERROR_NO_MEMORY;
         }
-        subscriptionIndex.mNodes[subscriptionIndex.mSize++] = subscriptionNode;
+        subscriptionIndex[subscriptionIndex.mSize++] = subscriptionNode;
         SaveIndex(subscriptionIndex);
     }
 
     // Load existing subscriptions for node, then combine and save state
     SubscriptionList subscriptions;
-    CHIP_ERROR err = FindByScopedNodeId(subscriptionNode, subscriptions);
+    CHIP_ERROR err = FindByScopedNodeId(subscriptionNode, subscriptions, 1);  // ask to allocate 1 extra space for new subscription
```

Comment:
What does "one extra space" mean?

Reply:
This code is no longer there due to the storage rework.